src/test/java/org/sagebionetworks/bridge/spring/controllers/ScheduledActivityControllerTest.java

```diff
@@ -0,0 +1,584 @@
+        DateTimeZone UNK = DateTimeZone.forOffsetHours(4);
+        StudyParticipant updatedParticipant = new StudyParticipant.Builder()
+                .copyOf(session.getParticipant())
+                .withTimeZone(UNK).build();
+        session.setParticipant(updatedParticipant);
+        
+        controller.getScheduledActivities(null, "-07:00", "3", "5");
+        
+        verify(mockScheduledActivityService).getScheduledActivities(eq(STUDY), contextCaptor.capture());
+        ScheduleContext context = contextCaptor.getValue();
+        assertEquals(context.getInitialTimeZone(), UNK);
+    }
+    
+    @SuppressWarnings("deprecation")
+    @Test
+    public void utcTimeZoneParsedCorrectly() throws Exception {
+        controller.getScheduledActivities(null, "+0:00", "3", "5");
+        
+        verify(mockScheduledActivityService).getScheduledActivities(eq(STUDY), contextCaptor.capture());
+        ScheduleContext context = contextCaptor.getValue();
+        assertEquals(DateUtils.timeZoneToOffsetString(context.getInitialTimeZone()), "+00:00");
+    }
+    
+    @SuppressWarnings("deprecation")
+    @Test
+    public void getScheduledActivtiesAssemblesCorrectContext() throws Exception {
+        mockEditAccount(mockAccountDao, mockAccount);
+        DateTimeZone MSK = DateTimeZone.forOffsetHours(3);
+        List<ScheduledActivity> list = ImmutableList.of();
+        when(mockScheduledActivityService.getScheduledActivities(eq(STUDY), any(ScheduleContext.class))).thenReturn(list);
+        
+        controller.getScheduledActivities(null, "+03:00", "3", "5");
+        
+        verify(mockScheduledActivityService).getScheduledActivities(eq(STUDY), contextCaptor.capture());
+        
+        ScheduleContext context = contextCaptor.getValue();
+        assertEquals(context.getInitialTimeZone(), MSK);
+        assertEquals(context.getCriteriaContext().getUserDataGroups(), USER_DATA_GROUPS);
+        assertEquals(context.getMinimumPerSchedule(), 5);
+        
+        CriteriaContext critContext = context.getCriteriaContext();
+        assertEquals(critContext.getHealthCode(), HEALTH_CODE);
+        assertEquals(critContext.getLanguages(), LANGUAGES);
+        assertEquals(critContext.getUserSubstudyIds(), USER_SUBSTUDY_IDS);
+        assertEquals(critContext.getStudyIdentifier().getIdentifier(), TEST_STUDY_IDENTIFIER);
+        assertEquals(critContext.getClientInfo(), CLIENT_INFO);
+        
+        verify(mockCacheProvider).updateRequestInfo(requestInfoCaptor.capture());
+        RequestInfo requestInfo = requestInfoCaptor.getValue();
+        assertEquals(requestInfo.getUserId(), "id");
+        assertEquals(requestInfo.getLanguages(), LANGUAGES);
+        assertEquals(requestInfo.getUserDataGroups(), USER_DATA_GROUPS);
+        assertEquals(requestInfo.getUserSubstudyIds(), USER_SUBSTUDY_IDS);
+        assertEquals(requestInfo.getStudyIdentifier(), TEST_STUDY);
+        assertNotNull(requestInfo.getActivitiesAccessedOn());
```

Comment:
Need to check requestInfo.getActivitiesAccessedOn().getZone()

Reply:
Added. This exposed what I think is a bug, so I changed the controller to get the desired behavior (it comes down to: what is the time zone in RequestInfo, is it the user's initial time zone, or is it the time zone of the most recent request? It looks like here, we're expecting it to be the latter, so I've changed the code so the test passes. This is now different than BridgePF.